pkg/orchestrator/etcd_worker.go

```diff
@@ -269,10 +312,22 @@ func (worker *EtcdWorker) applyPatches(ctx context.Context, patches []DataPatch)
 			op = clientv3.OpPut(key.String(), string(value))
 		} else {
 			op = clientv3.OpDelete(key.String())
+			hasDelete = true
 		}
 		ops = append(ops, op)
 	}
 
+	if hasDelete {
+		ops = append(ops, clientv3.OpPut(worker.prefix.String()+deletionCounterKey, fmt.Sprint(worker.deleteCounter+1)))
+	}
+	if worker.deleteCounter > 0 {
+		cmps = append(cmps, clientv3.Compare(clientv3.Value(worker.prefix.String()+deletionCounterKey), "=", fmt.Sprint(worker.deleteCounter)))
```

Comment:
What about `path.Join(worker.prefix.String(), deletionCounterKey)` ?

Reply:
We can do this. But the Etcd prefixes are all normalized here, so it must not end with `/`.